sandbox-data/src/main/java/com/exadel/sandbox/repository/event/specification/SpecificationBuilder.java

```diff
@@ -0,0 +1,113 @@
+package com.exadel.sandbox.repository.event.specification;
+
+import com.exadel.sandbox.model.vendorinfo.Event;
+import com.exadel.sandbox.model.vendorinfo.Status;
+import com.exadel.sandbox.repository.event.EventRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Repository;
+
+import java.util.Set;
+
+@Repository
+@RequiredArgsConstructor
+public class SpecificationBuilder {
+
+    private final EventRepository eventRepository;
+
+    public Page<Event> getEventsByParameters(Status status,
```

Comment:
If you have more than 3 parameters consider refactoring or wrapper object creation.
Effective Java, Chapter 7 (Methods), Item 40 (Design method signatures carefully)

Reply:
Yes, I understand this problem.
But EventFilterRequest is from the service level.
Will it be right if I pass this object as a method parameter from the service to the repo?